src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs

```diff
@@ -40,40 +41,44 @@ internal static partial class OrdinalCasing
         // s_casingTable is covering the Unicode BMP plane only. Surrogate casing is handled separately.
         // Every cell in the table is covering the casing of 256 characters in the BMP.
         // Every cell is array of 512 character for uppercasing mapping.
-        private static ushort []?[] s_casingTable =
+        private static ushort []?[]? s_casingTable;
+
+        // 0 - null
+        // 1 - s_noCasingPage
+        private static byte[] s_casingTableInit =
```

Comment:
Did you mean to use `bool[]` (saves 7 bits per element in this case)?

Reply:
A bool array still takes 1 byte per entry. It would be possible to store this is as an array of bitfields, but it would it make it much harder to read.

Reply:
Linked issue suggests `it needs to actually store only 32 bytes of data`, but it is 32x8. Maybe we can keep them as bytes and read bits with a mask. Plenty of similar code is doing bit twiddling, a comment will help clearing it up.